This is continuation of the already closed issue:
https://bugzilla.samba.org/show_bug.cgi?id=13704
so it will need a separate community issue to be raised.

diff --git a/source3/smbd/notify_msg.c b/source3/smbd/notify_msg.c
index ff38b96..1c6525f 100644
--- a/source3/smbd/notify_msg.c
+++ b/source3/smbd/notify_msg.c
@@ -111,7 +111,7 @@ static void notify_handler(struct messaging_context *msg, void *private_data,
 {
 	struct notify_context *ctx = talloc_get_type_abort(
 		private_data, struct notify_context);
-	struct notify_event_msg *event_msg;
+	struct notify_event_msg event_message, *event_msg = &event_message;
 	struct notify_event event;
 
 	if (data->length < offsetof(struct notify_event_msg, path) + 1) {
@@ -123,7 +123,8 @@ static void notify_handler(struct messaging_context *msg, void *private_data,
 		return;
 	}
 
-	event_msg = (struct notify_event_msg *)data->data;
+	/* avoid of SIGBUS */
+	memcpy((const void *)event_msg, (const void *)&data->data, sizeof(struct notify_event_msg));
 
 	event.action = event_msg->action;
 	event.path = event_msg->path;
diff --git a/source3/smbd/notifyd/notifyd.c b/source3/smbd/notifyd/notifyd.c
index 6ff1de5..26f7134 100644
--- a/source3/smbd/notifyd/notifyd.c
+++ b/source3/smbd/notifyd/notifyd.c
@@ -737,15 +737,19 @@ static void notifyd_trigger_parser(TDB_DATA key, TDB_DATA data,
 				   void *private_data)
 
 {
-	struct notifyd_trigger_state *tstate = private_data;
-	struct notify_event_msg msg = { .action = tstate->msg->action,
-					.when = tstate->msg->when };
+	struct notifyd_trigger_state loctstate, *tstate = &loctstate;
+	struct notify_event_msg msg;
 	struct iovec iov[2];
 	size_t path_len = key.dsize;
 	struct notifyd_instance *instances = NULL;
 	size_t num_instances = 0;
 	size_t i;
 
+	/* avoid SIGBUS */
+	memcpy(tstate, private_data, sizeof(loctstate));
+	msg.action = tstate->msg->action;
+	memcpy((const void *)&msg.when, (const void *)&tstate->msg->when, sizeof(msg.when));
+
 	if (!notifyd_parse_entry(data.dptr, data.dsize, &instances,
 				 &num_instances)) {
 		DEBUG(1, ("%s: Could not parse notifyd_entry\n", __func__));
